pkg/healthcheck/healthcheck_multicluster.go

```diff
@@ -54,108 +59,207 @@ var expectedServiceMirrorRemoteClusterPolicyVerbs = []string{
-				check: func(context.Context) error {
-					return hc.checkRemoteClusterConnectivity()
+				{
+					description: "service mirror controller can access target clusters",
+					hintAnchor:  "l5d-smc-target-clusters-access",
+					check: func(context.Context) error {
+						return hc.checkRemoteClusterConnectivity()
+					},
 				},
-			},
-			{
-				description: "all target cluster gateways are alive",
-				hintAnchor:  "l5d-multicluster-target-gateways-alive",
-				check: func(ctx context.Context) error {
-					return hc.checkRemoteClusterGatewaysHealth(ctx)
+				{
+					description: "all target cluster gateways are alive",
+					hintAnchor:  "l5d-multicluster-target-gateways-alive",
+					check: func(ctx context.Context) error {
+						return hc.checkRemoteClusterGatewaysHealth(ctx)
+					},
 				},
-			},
-			{
-				description: "clusters share trust anchors",
-				hintAnchor:  "l5d-multicluster-clusters-share-anchors",
-				check: func(ctx context.Context) error {
-					return hc.checkRemoteClusterAnchors()
+				{
+					description: "clusters share trust anchors",
+					hintAnchor:  "l5d-multicluster-clusters-share-anchors",
+					check: func(ctx context.Context) error {
+						return hc.checkRemoteClusterAnchors()
+					},
+				},
+				{
+					description: "multicluster daisy chaining is avoided",
+					hintAnchor:  "l5d-multicluster-daisy-chaining",
+					warning:     true,
+					check: func(ctx context.Context) error {
+						return hc.checkDaisyChains()
+					},
+				},
+				{
+					description: "all mirror services have endpoints",
+					hintAnchor:  "l5d-multicluster-services-endpoints",
+					warning:     true,
+					check: func(ctx context.Context) error {
+						return hc.checkIfMirroredServicesHaveEndpoints()
+					},
+				},
+				{
+					description: "all gateway mirrors have endpoints",
+					hintAnchor:  "l5d-multicluster-gateways-endpoints",
+					warning:     true,
+					check: func(ctx context.Context) error {
+						return hc.checkIfGatewayMirrorsHaveEndpoints()
+					},
+				},
+				{
+					description: "all mirror services have a valid gateway mirror",
+					hintAnchor:  "l5d-multicluster-mirror-gateways",
+					warning:     true,
+					check: func(ctx context.Context) error {
+						return hc.checkIfAllMirrorServicesHaveGatewayMirrors()
+					},
+				},
+				{
+					description: "remote: all gateways have external IPs",
+					hintAnchor:  "l5d-multicluster-gateways-exist",
+					warning:     true,
+					check: func(ctx context.Context) error {
+						return hc.checkRemoteIfGatewaysHaveExternalIP()
+					},
+				},
+				{
+					description: "remote: gateways exists for all exported services",
+					hintAnchor:  "l5d-multicluster-exported-gateways",
+					warning:     true,
+					check: func(ctx context.Context) error {
+						return hc.checkRemoteifGatewaysExistforExportedServices()
+					},
+				},
+				{
+					description: "remote: all gateways have correct port names exposed",
```

Comment:
I think all of these remote checks can be collapsed into a single check that checks the health of exported service on the remote cluster.  For each exported service on the remote cluster we should check that:

* the gateway service it points should exist
* that gateway should have an external IP
* that gateway should have the correct ports defined
* that gateway should have non-empty endpoints

Doing this all in one check makes logical sense to me and should simplify the code since we won't need to repeatedly fetch the list of gateways.